frontend/src/app/pages/exercises/exercise/shared/exercise-map/feature-managers/element-feature-manager.ts

```diff
@@ -167,15 +184,28 @@ export abstract class ElementFeatureManager<
         if (!isCoordinateArray(coordinates)) {
             return coordinates;
         }
+        if (!isCoordinateArrayOfArrays(coordinates)) {
+            // We need index 0 and 2 for the center
+            if (coordinates.length < 3) {
+                throw new Error(`Unexpectedly short array: ${coordinates}`);
+            }
+            return [
+                coordinates[0]![0]! +
+                    (coordinates[2]![0]! - coordinates[0]![0]!) / 2,
+                coordinates[0]![1]! +
+                    (coordinates[2]![1]! - coordinates[0]![1]!) / 2,
+            ];
+        }
+
         // We need index 0 and 2 for the center
-        if (coordinates.length < 3) {
-            throw new Error(`Unexpectedly short array: ${coordinates}`);
+        if (coordinates[0]!.length < 3) {
+            throw new Error(`Unexpectedly short array: ${coordinates[0]!}`);
         }
         return [
-            coordinates[0]![0]! +
-                (coordinates[2]![0]! - coordinates[0]![0]!) / 2,
-            coordinates[0]![1]! +
-                (coordinates[2]![1]! - coordinates[0]![1]!) / 2,
+            coordinates[0]![0]![0]! +
+                (coordinates[0]![2]![0]! - coordinates[0]![0]![0]!) / 2,
+            coordinates[0]![0]![1]! +
+                (coordinates[0]![2]![1]! - coordinates[0]![0]![1]!) / 2,
         ];
     }
```

Comment:
I'm a bit confused: could it be that we don't use this method (`getCenter()`) anywhere? If yes, then I would recommend removing it (YAGNI).

Reply:
I've removed it.